File: week-2/unique-morse-code-words/unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        hashtable = {}
        for word in words:
            transform = ''
            for letter in word:
                transform  += morse[ord(letter) - ord('a')]
            
            hashtable[transform] = True
            
        return len(hashtable.keys())

File: week-2/unique-morse-code-words/unique-morse-code-words.js
/**
 * @param {string[]} words
 * @return {number}
 */
var uniqueMorseRepresentations = function(words) {
    const morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."];
    const hashtable = {};
    for (let word of words) {
        let transform = '';
        for (let letter of word) {
            transform += morse[letter.charCodeAt() - 'a'.charCodeAt()]
        }
        
        hashtable[transform] = true;
    }
    
    return Object.keys(hashtable).length;
};
